evalml/pipelines/components/estimators/estimator.py

```diff
@@ -71,4 +71,4 @@ def feature_importance(self):
             raise MethodPropertyNotFoundError("Estimator requires a feature_importance property or a component_obj that implements feature_importances_")
 
     def __eq__(self, other):
-        return self.supported_problem_types == other.supported_problem_types and super().__eq__(other)
+        return super().__eq__(other) and self.supported_problem_types == other.supported_problem_types
```

Comment:
Small unrelated-ish change but necessary because if `other` is not an Estimator then `supported_problem_types` will raise an AttributeError.